Add boolean parsing in formulas

HactarCE opened the pull request:
This PR makes `TRUE` and `FALSE` work directly in formulas. So `IF(TRUE(), 1, 2)` is equivalent to `IF(TRUE, 1, 2)`. This is consistent with Excel.

davidkircos reviewed (changes requested):
@HactarCE functionality looks good! Should we update the docs to show TRUE?

It's also a little weird that TRUE autocompletes to TRUE(). Maybe we can turn that off? 

Also the tests fail.

<img width="237" alt="CleanShot 2023-06-22 at 13 39 03@2x" src="https://github.com/quadratichq/quadratic/assets/3479421/d8d9fd50-8faf-4ea6-a688-b06182b573b9">

HactarCE commented:
Good idea with the autocomplete snippet! I've implemented that, and fixed the tests (I made a mistake with parsing priority that I thought I had committed but hadn't)
